models/classes/TestSessionState/Exception/RestorationImpossibleException.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace oat\taoQtiTest\models\TestSessionState\Exception;
+
+class RestorationImpossibleException extends \Exception
```

Comment:
I would recommend having an internal project exception for a catch in a future only project depend exception exclude libraries ones

Reply:
didn't get it, could you provide little bit more context?

Reply:
understand, can't see right now useful cases, but why not, prefer to have on more view on this idea. @wazelin  WFYT

Reply:
Yeah, it sounds useful, but I honestly can't think of a single good use case as well 🤔

Anyway, should we need to group different exceptions in the future, a better way to do that IMO is to add an empty interface, and have your exceptions implement it. That way you can sorta _tag_ your exceptions, rather than put in a segregated group.